amazon-personalize-ranking/src/main/java/org/opensearch/search/relevance/transformer/personalizeintelligentranking/reranker/impl/AmazonPersonalizedRankerImpl.java

```diff
@@ -96,7 +99,15 @@ public SearchHits rerank(SearchHits hits, PersonalizeRequestParameters requestPa
 
             SearchHits personalizedHits = combineScores(hits, result);
             return personalizedHits;
-        } catch (Exception ex) {
+        } catch (AmazonServiceException e) {
+            logger.error("Exception while calling personalize campaign: {}", e.getMessage());
+            int statusCode = e.getStatusCode();
+            if (statusCode >= 400 && statusCode < 500) {
+                throw new IllegalArgumentException(INSUFFCIENT_PERMISSION_ERROR_MESSAGE);
```

Comment:
Can you please reconsider exception message provided for the 4xx or user error? This needs to be generic. I think you can consider using: ``new IllegalArgumentException(e);`` to propagate existing exception message to user.

Reply:
+1, shouldn't group all 4xx into the same exception message, the exception might be an unauthorized Exception from a 401, and insufficient permissions from a 403, but the users cannot tell from the same message.

Reply:
Updated in the revision